ScheduleLesson/Exception/CustomException.cs

```diff
@@ -0,0 +1,52 @@
+﻿namespace ScheduleLesson
+{
+    public class CustomException : Exception
+    {
+        public int StatusCode { get; }
+        public CustomException(string message, int statusCode = 500) : base(message)
+        {
+            StatusCode = statusCode;
+        }
+    }
+    public class ExceptionHandlingMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly ILogger<ExceptionHandlingMiddleware> _logger;
+
+        public ExceptionHandlingMiddleware(RequestDelegate next, ILogger<ExceptionHandlingMiddleware> logger)
+        {
+            _next = next;
+            _logger = logger;
+        }
+
+        public async Task InvokeAsync(HttpContext context)
+        {
+            try
+            {
+                await _next(context);
+            }
+            catch (CustomException ex)
+            {
+                _logger.LogError($"Custom error occurred: {ex.Message}");
+                await HandleExceptionAsync(context, ex.StatusCode, ex.Message);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError($"An unexpected error occurred: {ex.Message}");
```

Comment:
Where are the logs saved?

Reply:
The goal is to store these logs so they can be viewed, used, and analyzed later.

Reply:
When an exception occurs, the schedule is not saved anywhere. I don't understand why they should be saved when an exception occurs?